### DISCUSSION TITLE
Monitoring need also a sychronizing command?

### DISCUSSION THREAD

**Discussion by fohnbit** [abraunegg/onedrive#1282] [Q&A] [answered]
Hello,

when I start the daemon on ubuntu with **onedrive --monitoring**

Is this enough to sync in both directions? or did I need:  **onedrive --monitoring --synchronizing**?

Thank you!

**Comment by abraunegg** [comment, marked as answer]
@fohnbit 

To perform a sync in both directions + monitor local file systems for change + run constantly (either as a systemd service or other), the following is sufficient: `onedrive --monitor`

To perform a 1 time sync in both directions, the following is sufficient: `onedrive --synchronize`

To 'tweak' the sync operations (1-way as download or upload only) or other options, refer to the [usage document](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md).

**Reply by fohnbit** [reply]
@abraunegg 

Great thank you. I have so much files in my onedrive, so it takes just a lot of time, after a sync is started.

Its now working

**Reply by abraunegg** [reply]
@fohnbit 
Just remember OneDrive has a *real* problem with large numbers of files. If you have more than 100,000 local files that will be in scope for syncing, you should consider using selecting sync to *only* sync what you need to sync to OneDrive. Any more than 100K files, OneDrive will be *very* slow and have huge performance issues. This is a OneDrive API / OneDrive implementation issue by Microsoft, not this client.

Reference: https://support.microsoft.com/en-us/office/invalid-file-names-and-file-types-in-onedrive-and-sharepoint-64883a5d-228e-48f5-b3d2-eb39e07630fa?ui=en-us&rs=en-us&ad=us#synctoomany
